java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/MultiCryptoProviderDecryptionTest.java

```diff
@@ -0,0 +1,211 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client.crypto;
+
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.client.MantaObjectInputStream;
+import com.joyent.manta.client.MantaObjectResponse;
+import com.joyent.manta.http.MantaHttpHeaders;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.collections4.map.CaseInsensitiveMap;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.BoundedInputStream;
+import org.apache.commons.lang3.Validate;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.conn.EofSensorInputStream;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.testng.annotations.Optional;
+
+import static com.joyent.manta.client.crypto.ExternalSecurityProviderLoader.*;
+import static com.joyent.manta.http.MantaHttpHeaders.*;
+import static org.apache.http.HttpHeaders.*;
+import static org.apache.commons.codec.binary.Hex.decodeHex;
+
+import javax.crypto.SecretKey;
+import java.io.*;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.Provider;
+import java.util.*;
+
+@Test(groups={"encryption-provider"})
+public class MultiCryptoProviderDecryptionTest {
+    private static final Logger LOG = LoggerFactory.getLogger(MultiCryptoProviderDecryptionTest.class);
+    private static final List<SupportedCipherDetails> CIPHERS_TO_TEST = Collections.unmodifiableList(Arrays.asList(
+                AesGcmCipherDetails.INSTANCE_128_BIT,
+                AesGcmCipherDetails.INSTANCE_192_BIT,
+                AesGcmCipherDetails.INSTANCE_256_BIT,
+                AesCtrCipherDetails.INSTANCE_128_BIT,
+                AesCtrCipherDetails.INSTANCE_192_BIT,
+                AesCtrCipherDetails.INSTANCE_256_BIT,
+                AesCbcCipherDetails.INSTANCE_128_BIT,
+                AesCbcCipherDetails.INSTANCE_192_BIT,
+                AesCbcCipherDetails.INSTANCE_256_BIT));
+
+    private byte[] expectedContent;
+
+    @BeforeClass()
+    @Parameters({"preferredProviders"})
+    public void beforeClass(@Optional final String preferredProviders) throws IOException {
+        /* Warn user to invoke it from the TestNG suite */
+        Validate.notNull(preferredProviders,
+                    "Invoke value of parameter preferredProviders from the TestNG Suite!");
+
+        /* Backup provider list from static context */
+        List<Provider> providers = buildRankedPreferredProviders(preferredProviders);
+        setRankedPreferredProviders(providers);
+        final URL testURL = getClass().getClassLoader().getResource("test-data/chaucer.txt");
+
+        try (InputStream in = testURL.openStream()) {
+            expectedContent = IOUtils.toByteArray(in);
+        } catch (NullPointerException e) {
+            LOG.warn("Original Data from resources for" +
```

Comment:
You handled the exception and warned. However, this means that the program will continue to run even though you had a `NullPointerException`. This is typically a bad thing. Perhaps add:
```java
throw e;
```

Reply:
Done.